doc/tutorials/imgproc/basic_geometric_drawing/basic_geometric_drawing.markdown

```diff
@@ -82,20 +81,20 @@ Code
 
 @add_toggle_cpp
 -   This code is in your OpenCV sample folder. Otherwise you can grab it from
-    [here](https://raw.githubusercontent.com/opencv/opencv/3.4/samples/cpp/tutorial_code/core/Matrix/Drawing_1.cpp)
-    @include samples/cpp/tutorial_code/core/Matrix/Drawing_1.cpp
+    [here](https://raw.githubusercontent.com/opencv/opencv/3.4/samples/cpp/tutorial_code/ImgProc/basic_drawing/Drawing_1.cpp)
+    @include samples/cpp/tutorial_code/ImgProc/basic_drawing/Drawing_1.cpp
```

Comment:
BTW, Such renames breaks direct links of already released documentation (for example, 3.4.2 points to old 3.4 path).

Currently I don't have right approach how to work with this:
- we don't want to update/rebuild old documentation
- fix tag 3.4 -> 3.4.2 doesn't make sense (file is already part of documentation via `@include` statements). Also we prefer to keep "version increment" commit simple as possible.

Suggestions are welcome.

Reply:
I think we can add an alias to Doxyfile:
```
ALIASES               += github{1}="https://github.com/opencv/opencv/blob/@CMAKE_DOXYGEN_GIT_COMMIT@/\1'"
```
Then get current commit hash in docs/CMakeLists.txt:
```.cmake
if(GIT_FOUND)
  execute_process(COMMAND "${GIT_EXECUTABLE}" rev-parse HEAD
    WORKING_DIRECTORY "${OpenCV_SOURCE_DIR}"
    OUTPUT_VARIABLE CMAKE_DOXYGEN_GIT_COMMIT
    ERROR_QUIET
    OUTPUT_STRIP_TRAILING_WHITESPACE
  )
else()
  set(CMAKE_DOXYGEN_GIT_COMMIT "3.4")
endif()
```
And then use this alias:
```
**Downloadable code**: Click [here](@github{samples/cpp/tutorial_code/ImgProc/BasicLinearTransforms.cpp})
```

Reply:
Problem here is that such links are usually provided by this wording: "the latest code can be found here".

Also, It is not necessary to link on the current code - it is already attached into documentation in general.